pkgs/development/python-modules/click-option-group/default.nix

```diff
@@ -1,32 +1,40 @@
 {
   lib,
   buildPythonPackage,
-  pythonOlder,
   fetchFromGitHub,
+  setuptools,
   click,
   pytestCheckHook,
 }:
 
 buildPythonPackage rec {
   pname = "click-option-group";
   version = "0.5.6";
-  format = "setuptools";
-  disabled = pythonOlder "3.6";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "click-contrib";
-    repo = pname;
+    repo = "click-option-group";
```

Comment:
This change seems unnecessary, since the variable happens to be the same.

Re-using `pname` via `rec` is not like using `finalAttrs.pname`, which would change when overridden.

Reply:
It is discouraged to inherit it from `finalAttrs`, since that could cause issues when _overriding_ the `pname`` attr.

I can't see how inheriting via `rec` would ever lead to silent errors; you would get a 404 and/or hash mismatch if the `repo` argument became invalid.